Formula/b/bigquery-emulator.rb

```diff
@@ -0,0 +1,25 @@
+class BigqueryEmulator < Formula
+  desc "Emulate a GCP BigQuery server on your local machine"
+  homepage "https://github.com/goccy/bigquery-emulator"
+  url "https://github.com/goccy/bigquery-emulator.git",
+    tag:      "v0.6.1",
+    revision: "316038b8dd3f6534a87948f28ac54c41dc6afc21"
+  license "MIT"
+  head "https://github.com/goccy/bigquery-emulator.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "1"
+    ENV.cxx11
+
+    static_link_flags = '-linkmode external -extldflags "-static"' if OS.linux?
+
+    ldflags = "-s -w -X main.version=#{version} -X main.revision=#{revision} #{static_link_flags}"
+    system "go", "build", *std_go_args(ldflags:), "./cmd/bigquery-emulator"
+  end
+
+  test do
+    assert_match "version: #{version} (#{revision})", shell_output("#{bin}/bigquery-emulator --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @chenrui333, thanks for the speedy review. I've added better test and reviewed the Linux installation process.